### PR TITLE
Create new UI version of simple sign in page

### PR DESCRIPTION
**Pull request by TurnerRiley**
This PR updates the styling of the simple sign-up form (currently just used for LMS) to look like [this Figma](https://www.figma.com/design/Km9JCLNbayBlTrlnKenumq/New-PL-Program?node-id=2326-4536&t=zcUcuSKw2wUnaFXy-0).

One note about styling is that the "Cancel" button link at the bottom was not present in the Figmas (neither the original nor new) so I wasn't sure if I should remove it or not. I kept it in and added some padding but am open to making any updates to it!

lms

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-2027)
Followup ticket to integrate this with the workshop enroll form: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-2028)

## Testing story
Local testing.

## Followup work
Create generalized wrapper that will either use the current `LtiLinkAccountPage` (for LMS) or will use a new `WorkshopAccountPage` (for workshop content).

### CODE REVIEW

**Comment by moshebaricdo** [comment]
Looking goodddd! A couple minor things I'm seeing:
- For the buttons in the cards, large is being pulled for the size prop, let's update to be medium
- For the teal icons in the card, size should be 2.5x instead of 3x
- There's a big gap between the card titles and the body copy. Body copy should be more or less equal vertically between the button and the card title. But looks like it's pulling from styling outside of the css specific to this page.
- Also for cards, let's try like 4px more spacing between the icon and the card title.
- Let's make the margin-top for "cdo-logo" 64px to match the spacing between the header and the cards.
- Last one! bg should be the 50 gray value (and cards white), we'll have the same styling for sign-up flow pages as well!

File: apps/i18n/common/en_us.json [hunk 1]
```diff
@@ -1846,6 +1846,11 @@
   "noLibrariesInClass": "No one in your class has published a library. Try adding one from an ID.",
   "noMenuItemsAvailable": "No menu items available.",
   "nominateATeacher": "Nominate a Teacher",
+  "nonLMSLinkAccountWelcomeBannerHeaderLabel": "Continue with a Code.org account",
+  "nonLMSLinkAccountWelcomeBannerContentWorkshopEnroll": "To enroll in workshops, you’ll need a Code.org educator account.",
+  "nonLMSAccountExistingAccountCardContentWorkshopEnroll": "You’ll need to sign in to your Code.org account to enroll in this workshop",
+  "nonLMSAccountNewAccountCardContentWorkshopEnroll": "You’ll need to create an account on Code.org to enroll in this workshop",
+  "nonLMSAccountNewAccountCardActionLabel": "Create an account",
```
Comment (bethanyaconnor) [inline apps/i18n/common/en_us.json]: nit: eventually, we might extend this page to more non-LMS scenarios. Wondering if some of these would benefit from a key that doesn't mention LMS at all.

For example, `nonLMSAccountNewAccountCardActionLabel` could be `createAnAccount` (follows the pattern we've used for short strings in the past). `nonLMSAccountExistingAccountCardContentWorkshopEnroll` could be `accountExistingAccountCardContentWorkshopEnroll`

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx
Comment (bethanyaconnor) [inline apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx:1]: Can you make a ticket to move all of this out of the LTI folders? If we're going to use this across scenarios, I'd prefer for this to be in a more generic spot :)

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ExistingAccountCard/index.tsx [hunk 2]
```diff
@@ -24,6 +24,7 @@ const ExistingAccountCard = () => {
     emailAddress,
     userType,
   } = useContext(LtiProviderContext)!;
+  const isLMS = true;
```
Comment (bethanyaconnor) [inline apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ExistingAccountCard/index.tsx]: Why is this needed? It doesn't appear to change anywhere?

Comment (bethanyaconnor) [reply apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ExistingAccountCard/index.tsx]: (Same question in all of the files below)

Comment (bethanyaconnor) [reply apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ExistingAccountCard/index.tsx]: 🤦‍♀️ I see in your description now that you'll update as a follow-up!

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx [hunk 3]
```diff
@@ -1,54 +1,37 @@
...
-
-      <Typography semanticTag={'p'} visualAppearance={'body-two'}>
-        {i18n.ltiLinkAccountWelcomeBannerContent({
-          providerName: ltiProviderName,
-        })}
-      </Typography>
+      <img className={styles.cdoLogo} src={codeLogo} alt={i18n.codeLogo()} />
+
+      <div className={styles.titleContainer}>
+        <Typography semanticTag={'h1'} visualAppearance={'heading-xxl'}>
```
Comment (bethanyaconnor) [inline apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx:18]: nit: use `<Heading1 visualAppearance={'heading-xxl'}>` here

**Review by bethanyaconnor — COMMENTED** [review]
Got some future-proofing comments but other than the string keys, none block this PR. Thanks for staging this out!

File: apps/i18n/common/en_us.json (hunk 1 above)
Comment (TurnerRiley) [reply apps/i18n/common/en_us.json]: Oooo good point! On it!

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx
Comment (TurnerRiley) [reply apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx:1]: Will do!

Comment (TurnerRiley) [reply apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx:1]: Tracked here: https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-2065

**Review by moshebaricdo — APPROVED** [review]

**Review by moshebaricdo — APPROVED** [review]
Looks good to me!! Thank you!!

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx [hunk 4]
```diff
@@ -1,54 +1,37 @@
...
-      {ltiProviderIcon && (
-        <div className={styles.welcomeBannerContainer}>
-          <img src={ltiProviderIcon} alt={ltiProviderName} />
-          <FontAwesomeV6Icon
-            className={styles.exchangeIcon}
-            iconName={'exchange'}
-          />
-          <img src={codeLogo} alt={i18n.codeLogo()} />
-        </div>
-      )}
```
Comment (stephenliang) [inline apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx]: Is it intentional for the LMS <-> Code.org images to be removed?

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx [hunk 5]
```diff
@@ -1,54 +1,37 @@
...
-      case 'schoology':
-        return schoology;
-      default:
-        return undefined;
-    }
-  }, [ltiProvider]);
-
-  const ltiProviderIcon = getLtiProviderIcon();
+  const {ltiProviderName} = useContext(LtiProviderContext)!;
+  const isLMS = true;
```
Comment (stephenliang) [inline apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx]: This is for later, not necessarily as part of this PR, but I wanted to check your thoughts on when you circle back to refactor LMS vs non-LMS, I think we can use some compositional branching logic for all of the components in this directory (perhaps generalizing this page completely), I'm thinking of something like:

`WelcomeBanner` switches to `LtiWelcomeBanner` or `ProfessionalDevelopmentWelcomeBanner` depending on the type.

I'd like to avoid having a single WelcomeBanner component that encapsulates all the different types. (This comment applies to all the components)

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx [hunk 6]
```diff
@@ -1,54 +1,37 @@
...
-      case 'canvas_test_cloud':
-        return canvas;
-      case 'schoology':
-        return schoology;
-      default:
-        return undefined;
-    }
-  }, [ltiProvider]);
-
-  const ltiProviderIcon = getLtiProviderIcon();
```
Comment (stephenliang) [inline apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx]: Is it intentional for the LMS <-> Code.org images to be removed?

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ContinueCard/index.tsx [hunk 7]
```diff
@@ -57,19 +57,21 @@ const ContinueAccountCard = () => {
...
         }
       />
-      <CardContent>
-        {i18n.ltiLinkAccountContinueAccountCardContent()}
-      </CardContent>
+      <div className={classNames(styles.cardContentContainer)}>
+        <CardContent>
+          {i18n.ltiLinkAccountContinueAccountCardContent()}
+        </CardContent>
+      </div>
```
Comment (stephenliang) [inline apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ContinueCard/index.tsx]: I haven't tested this, but I believe you can have the `cardContentContainer` class be passed into `CardContent` directly and avoid the parent div.

```suggestion
      <CardContent className={classNames(styles.cardContentContainer)}>
         {i18n.ltiLinkAccountContinueAccountCardContent()}
      </CardContent>
```

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ExistingAccountCard/index.tsx [hunk 8]
```diff
@@ -51,22 +52,26 @@ const ExistingAccountCard = () => {
...
-      </CardContent>
+      <div className={classNames(styles.cardContentContainer)}>
+        <CardContent className={cardStyles.cardContent}>
+          {isLMS
+            ? i18n.ltiLinkAccountExistingAccountCardContent({
+                providerName: ltiProviderName,
+              })
+            : i18n.accountExistingAccountCardContentWorkshopEnroll()}
+        </CardContent>
+      </div>
```
Comment (stephenliang) [inline apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ExistingAccountCard/index.tsx]: I haven't tested this, but I believe you can have the padding be added to the CardContent div

```suggestion
      <CardContent className={classNames(styles.cardContentContainer, cardStyles.cardContent)}>
        {isLMS
          ? i18n.ltiLinkAccountExistingAccountCardContent({
              providerName: ltiProviderName,
            })
          : i18n.accountExistingAccountCardContentWorkshopEnroll()}
      </CardContent>
```

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/NewAccountCard/index.tsx [hunk 9]
```diff
@@ -67,33 +68,41 @@ const NewAccountCard = () => {
...
+            className={classNames(styles.icon, 'fa-2x')}
             iconName={'user-plus'}
           />
         }
       />
-      <CardContent>
-        {i18n.ltiLinkAccountNewAccountCardContent({
-          providerName: ltiProviderName,
-        })}
+      <div className={classNames(styles.cardContentContainer)}>
```
Comment (stephenliang) [inline apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/NewAccountCard/index.tsx]: I believe `cardContentContainer` can be passed as a class to `CardContent`

**Comment by stephenliang** [comment]
As we appear to be generalizing the LTI Account Linking disambiguation page, what do you think of either:

1. Generalize the LTI page to a generic disambiguation page, and based on the type we render out the LTI page or the workshop page.
2. Clone the LTI page to a new page for workshops

I'd like to use the compositional pattern as much here, as it appears we are creating variants (with more coming soon?) of this page

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ContinueCard/index.tsx (hunk 7 above)
Comment (TurnerRiley) [reply apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ContinueCard/index.tsx]: Oh I had tried to pass them in but ran into some typescript errors (though I'm new to typescript so I may have just been implementing wrong)!

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx (hunk 6 above)
Comment (TurnerRiley) [reply apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx]: Yes, this was part of the Figma but I can definitely preserve this with going the route of splitting this up

**Comment by TurnerRiley** [comment]
> As we appear to be generalizing the LTI Account Linking disambiguation page, what do you think of either:
> 
> 1. Generalize the LTI page to a generic disambiguation page, and based on the type we render out the LTI page or the workshop page.
> 2. Clone the LTI page to a new page for workshops
> 
> I'd like to use the compositional pattern as much here, as it appears we are creating variants (with more coming soon?) of this page

@dmcavoy Looking at the next task of implementing the simple sign-up flow, there would have to be some refactoring anyways since the workshop use of it wouldn't need any of the redux attributes. I like Stephen's ideas here and am curious about your thoughts here? Bethany requested a followup ticket of moving these files to a more shared area (or at the very least outside the LTI folders) since we were intending to use this in multiple places, so that could be a good time to either clone or generalize.

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ContinueCard/index.tsx (hunk 7 above)
Comment (TurnerRiley) [reply apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ContinueCard/index.tsx]: I can definitely dig deeper but want to hear back about what approach we'll take regarding the generalization vs. cloning discussion so I'll hold off on further updates until that's figured out.

File: apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx (hunk 6 above)
Comment (stephenliang) [reply apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx]: @moshebaricdo Just want to confirm that this is to be intentionally removed. If so, I think we can remove the icons from the repo as well.

Comment (moshebaricdo) [reply apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx]: Yes, the LMS view should preserve the two icons.

Comment (TurnerRiley) [reply apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx]: Oh I think I misunderstood the Figma, so we want to keep the arrows and the icons for the LMS view? I thought we were removing them alltogether

Comment (moshebaricdo) [reply apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx]: Ah yes, sorry, there is a separate frame for the LMS version in the mocks! There should be some redlines for spacing between them

Comment (TurnerRiley) [reply apps/src/lib/ui/lti/link/LtiLinkAccountPage/WelcomeBanner/index.tsx]: Oh I see now, no my bad for not seeing the separation there. I'll fix that then!

**Comment by TurnerRiley** [comment]
> > As we appear to be generalizing the LTI Account Linking disambiguation page, what do you think of either:
> > 
> > 1. Generalize the LTI page to a generic disambiguation page, and based on the type we render out the LTI page or the workshop page.
> > 2. Clone the LTI page to a new page for workshops
> > 
> > I'd like to use the compositional pattern as much here, as it appears we are creating variants (with more coming soon?) of this page
> 
> @dmcavoy Looking at the next task of implementing the simple sign-up flow, there would have to be some refactoring anyways since the workshop use of it wouldn't need any of the redux attributes. I like Stephen's ideas here and am curious about your thoughts here? Bethany requested a followup ticket of moving these files to a more shared area (or at the very least outside the LTI folders) since we were intending to use this in multiple places, so that could be a good time to either clone or generalize.

We discussed offline and decided to go with option 1 of generalizing the LTI page to a generic disambiguation page! I'll update this PR to just be the styling updates, then the followup work of actually implementing the workshop version will be tracked [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-2028).

**Comment by carl-codeorg** [comment]
Read through the comments and I agree with the decision to make it generic and conditionally render different components for the PL vs LTI scenarios.

I did have one question about the styling though: is there supposed to be padding on the sides of the light gray div? Here's what it looks like locally for me:

Screenshot 2024-07-15 at 2 39 34 PM

The cancel button and the sides of the two cards look a bit too close to the edges IMO.

**Comment by TurnerRiley** [comment]
> Read through the comments and I agree with the decision to make it generic and conditionally render different components for the PL vs LTI scenarios.
> 
> I did have one question about the styling though: is there supposed to be padding on the sides of the light gray div? Here's what it looks like locally for me:
> 
> Screenshot 2024-07-15 at 2 39 34 PM
> The cancel button and the sides of the two cards look a bit too close to the edges IMO.

Oh thanks for catching this! I may have had my window too small so I didn't see the odd edges. Let me see if I can repro and I'll update!

Edit: Fixed!

**Review by carl-codeorg — APPROVED** [review]
Thanks for fixing up the background. Looks good now locally!
